host/frontend/host-orchestrator/orchestrator/instancemanager.go

```diff
@@ -112,17 +112,21 @@ type ProcedureBuilder interface {
 }
 
 type LaunchCVDProcedureBuilder struct {
+	paths                   IMPaths
 	stageDownloadCVD        *StageDownloadCVD
 	stageStartCVDServer     *StageStartCVDServer
 	stageCreateArtifactsDir *StageCreateDirIfNotExist
 	stageCreateHomesDir     *StageCreateDirIfNotExist
+	stageFetchCVDMap        map[string]*StageFetchCVD
```

Comment:
I had a hard time understanding why this is a map. Could you add a comment explaining its purpose? Something like:
```
// FetchCVD stages can be reused to avoid multiple downloads of the same artifacts
```
Also, calling a cache instead of a Map will probably make its function clearer (map is an implementation detail)

Reply:
Acknowledged, added a relevant comment.